### PR TITLE
Added protocol ID provider checker

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added functionality that can provide if a provided pid has a compatible protocol ID with the current network messenger instance

### CODE REVIEW

File: p2p/libp2p/connectionsHandler.go [hunk 1]
```diff
@@ -129,6 +133,25 @@ func (handler *connectionsHandler) Peers() []core.PeerID {
...
+	protocols, err := handler.p2pHost.Peerstore().GetProtocols(peer.ID(pid))
+	if err != nil {
+		handler.log.Trace("connectionsHandler.HasCompatibleProtocolID error getting protocols",
+			"pid", pid.Pretty(), "error", err.Error())
+		return false
+	}
+
+	for i := 0; i < len(protocols); i++ {
+		// this suffix is done automatically in dht.go L280 (v1proto := cfg.ProtocolPrefix + kad1)
+		if handler.protocolID+kadProtocol == string(protocols[i]) {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply p2p/libp2p/connectionsHandler.go:154]: can not return here as the first protocol that does not match will make the function return false. 
Also, an unreachable code on the for loop is shown by the linter and also the `GetProtocols returns a matching protocol ID` test will fail.

[the remark this answers is not in the document]
Comment (sstanculeanu) [reply p2p/libp2p/connectionsHandler.go:154]: Right 🤦

**Review by sstanculeanu — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

File: p2p/libp2p/connectionsHandler.go [hunk 2]
```diff
@@ -16,6 +16,7 @@ import (
 )
 
 const timeBetweenPeerPrints = time.Second * 20
+const kadProtocol = "/kad/1.0.0"
```
Comment (AdoAdoAdo) [inline p2p/libp2p/connectionsHandler.go:19]: shouldn't this be taken from the config?

Comment (iulianpascalau) [reply p2p/libp2p/connectionsHandler.go:19]: it is taken from config the protocol ID, this is just a suffix that the libp2p library adds it automatically

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**